config/config.go

```diff
@@ -113,17 +112,46 @@ func defaultConfigData() *configurationData {
 
 // NewConfig returns a configuration
 func NewConfig() (*Config, error) {
-	filename := os.Getenv("SNOWBRIDGE_CONFIG_FILE")
-	if filename == "" {
-		return newEnvConfig()
-	}
 
-	switch suffix := strings.ToLower(filepath.Ext(filename)); suffix {
-	case ".hcl":
-		return newHclConfig(filename)
+	var config *Config
+	var err error
+
+	// If we don't have a config file path, use env var configuration.
+	// Otherwise, use the file.
+	switch filename := os.Getenv("SNOWBRIDGE_CONFIG_FILE"); filename {
+	case "":
+		config, err = newEnvConfig()
+		if err != nil {
+			return nil, errors.Wrap(err, "Error parsing env config")
+		}
+
+		// If we have set TRANSFORM_CONFIG_B64, parse it into a config and overwrite the Transformation portion of our existing config.
+		if b64Transformations := os.Getenv("TRANSFORM_CONFIG_B64"); b64Transformations != "" {
+			transformationEnvConfig, err := base64.StdEncoding.DecodeString(b64Transformations)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error decoding b64 data from TRANSFORM_CONFIG_B64 env var")
+			}
+
+			transformationConfig, err := newHclConfig(transformationEnvConfig, "")
+			if err != nil {
+				return nil, errors.Wrap(err, "Error creating config fom provided TRANSFORM_CONFIG_B64")
+			}
+
+			config.Data.Transformations = transformationConfig.Data.Transformations
```

Comment:
Not faulting you for this but its one thing that always sort of bothers me about just overriding values in structs in this very mutable way.  Not sure if "setter" functions make the situation much better as its ultimately the same thing but direct editing struct values just feels .. bad.

Reply:
Yeah no I do actually agree - it's just lots more effort to do it not bad, have to wrangle with the underlying parser stuff which is more complex than I'd like it to be in the first place.

Ultimately we plan on ripping this logic back out regardless so I settled for this. Long term I'd love to both rip out the env vars (relying on only hcl env var functionality for it), and also simplify configuration and config parsing greatly.